src/RearOutlanderinverter.cpp

```diff
@@ -23,89 +23,93 @@
 
 RearOutlanderInverter::RearOutlanderInverter()
 {
-   //ctor
+    //ctor
 }
 
 void RearOutlanderInverter::SetCanInterface(CanHardware* c)
 {
-   can = c;
+    can = c;
 
-   can->RegisterUserMessage(0x289);//Outlander Inv Msg
-   can->RegisterUserMessage(0x299);//Outlander Inv Msg
-   can->RegisterUserMessage(0x733);//Outlander Inv Msg
+    can->RegisterUserMessage(0x289);//Outlander Inv Msg
+    can->RegisterUserMessage(0x299);//Outlander Inv Msg
+    can->RegisterUserMessage(0x733);//Outlander Inv Msg
 }
 
 void RearOutlanderInverter::DecodeCAN(int id, uint32_t data[2])
 {
-   uint8_t* bytes = (uint8_t*)data;// arrgghhh this converts the two 32bit array into bytes. See comments are useful:)
-   switch (id)
-   {
-   case 0x289:
-      //speed = (((data[0] >> 8)& 0xFF00) | ((data[0] >> 24) & 0x0FF)) - 20000;
-      //voltage = ((data[1] & 0xFF) << 8) | ((data[1] >> 8) & 0xFF);
-      speed = (bytes[2] * 256 | bytes[3]) - 20000;
-      voltage = (bytes[4] * 256) + bytes[5];
-      break;
-   case 0x299:
-      //motor_temp = bytes[0] -40;
-      inv_temp = ((bytes[1]-40) + (bytes[4] - 40)) / 2;
-      break;
-   case 0x733:
-      motor_temp = bytes[0] -40; //
-      break;
-
-
-   }
+    uint8_t* bytes = (uint8_t*)data;// arrgghhh this converts the two 32bit array into bytes. See comments are useful:)
+    switch (id)
+    {
+    case 0x289:
+        //speed = (((data[0] >> 8)& 0xFF00) | ((data[0] >> 24) & 0x0FF)) - 20000;
+        //voltage = ((data[1] & 0xFF) << 8) | ((data[1] >> 8) & 0xFF);
+        speed = (bytes[2] * 256 | bytes[3]) - 20000;
+        voltage = (bytes[4] * 256) + bytes[5];
+        break;
+    case 0x299:
+        //motor_temp = bytes[0] -40;
+        inv_temp = ((bytes[1]-40) + (bytes[4] - 40)) / 2;
+        break;
+    case 0x733:
+        motor_temp = bytes[0] -40; //
+        break;
+
+
+    }
 }
 
 
 
 void RearOutlanderInverter::SetTorque(float torquePercent)
 {
-   final_torque_request = (((torquePercent * 2000) / 100.0f ) * -1) + 10000; // *-1 reverses torque direction
-
-
-   Param::SetInt(Param::torque,final_torque_request);//post processed final torque value sent to inv to web interface
+    final_torque_request = ((torquePercent * 2000) / 100.0f ); //!! Moved into parameter *-1 reverses torque direction
+
+    if(Param::GetInt(Param::reversemotor) == 1)
+    {
+        final_torque_request *= -1;//reverse torque request to flip motor rotation
+    }
+    final_torque_request += 10000;
```

Comment:
I don't quite get that line. It's not there for the leaf inverter.

Why potentially making negative torque only to add 10000 to it? 🤔

Reply:
For creating the right value to writing to CAN for the inverter. It has a 10000 offset,

Reply:
thx, i now of course see the +10000 above in the original line.

i guess that's why code reviews on github mobile are not a good idea 🤦‍♂️